Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -1882,14 +1882,17 @@ static inline bool is_valid_name_character(u32 code_point)
 
 bool Document::is_valid_name(DeprecatedString const& name)
 {
-    if (name.is_empty())
+    auto code_points = Utf8View { name };
+    auto it = code_points.begin();
+    if (code_points.is_empty())
         return false;
 
-    if (!is_valid_name_start_character(name[0]))
+    if (!is_valid_name_start_character(*it))
         return false;
+    ++it;
```

Comment:
Am I missing something, or why are we checking the first `it` separately?

Reply:
it seems that the start character has slightly different logic that subsequent ones…

it seems that the `is_valid_name_character` method name could be improved then! 😅

Reply:
Got it, missed `_start_` in the first character check